File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def getDivisorCount(num):
    if num == 1: return 1

    count = 0
    for i in range(1, int(num**0.5)+1):
        if num % i == 0:
            count += 1
            if (num / i != i): count += 1
    return count

def solution(number, limit, power):
    arr = []
    for i in range(1, number+1):
        if getDivisorCount(i) > limit:
            arr.append(power)
        else:
            arr.append(getDivisorCount(i))
    
    return sum(arr)

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function getDivisorCount(num) {
    if (num === 1) return 1;
    
    let count = 0;
    for (let i = 1; i <= Math.sqrt(num); i++) {
        if (num % i === 0){
            count++;
            if (num / i !== i) count++;
        }
    }
    return count;
}

function solution(number, limit, power) {
    const arr = [];
    for (let i = 1; i < number + 1; i++){
        arr.push(getDivisorCount(i));
    }
    
    for (let i = 0; i < arr.length; i++) {
        if (arr[i] > limit) {
            arr[i] = power
        }
    }
    return arr.reduce((a, b) => a + b);
}